zuul-core/src/main/java/com/netflix/zuul/monitoring/ConnCounter.java

```diff
@@ -91,7 +108,8 @@ public void increment(String event) {
         lastCountKey = event;
         Id id = registry.createId(metricBase.name() + '.' + event).withTags(metricBase.tags()).withTags(dimTags);
         Gauge gauge = registry.gauge(id);
-        synchronized (gauge) {
+
+        synchronized (locks[id.hashCode() & LOCK_MASK]) {
```

Comment:
It's going to be a pain to look back at this in the future and try to understand what's going on here.

Reply:
I can split it into another method if you would like, but I dont know if I can avoid the lock until the issue (commented above) is fixed.